_drafts/2021-03-15-data_science_timeseries_plotting_notebook_template.md

```diff
@@ -0,0 +1,146 @@
+|:-----|-----------:|--------------:|
+| 0    | 2020-01-01 |         Honda |
+| 1    | 2020-02-01 |        Toyota |
+| 2    | 2020-01-01 |         Other |
+| ...  |        ... |           ... |
+
+[switrs_data]: {% post_url 2020-11-24-switrs_sqlite_hosted_dataset %}
+
+The [**time series plotting notebook**][plotting_nb] has two helpful functions
+to visualize this data.
+
+### Plot Time Series
+
+The first function, `plot_time_series()` is simple. It takes a dataframe
+formatted like the above data and returns a plot showing the number of events
+for each value in the categorical column. For example, to plot the number of
+accidents per week by vehicle make, we would call:
+
+```python
+plot_time_series(
+  df, ax,
+  date_col="datetime",
+  category_col="vehicle_make",
+  resample_frequency="W",  # Resample to 'W'eeks
+)
+```
+
+Which would produce this plot:
+
+[![A simple plot of the number of collisions by vehicle make in
+California][first_plot]][first_plot]
+
+[first_plot]: {{ file_dir }}/make_collision_in_time_first_version.svg
+
+The function has as few optional parameters:
+
+- `resample_frequency` which controls the timescale over which the data is
+aggregated.
+- `aggfunc` which controls how the data is aggregated.
+- `linewidth` which can be used to make the lines larger if there are only a
+few of them, or thinner if there is lots of data.
+
+### Simple Legend
+
+Simple legends are great; they convey their information most effectively
+because all the superfluous noise has been removed. 
+
+My [basic plotting notebook][first_notebook] has a function to remove all the
+extra information from the legend box leaving only the color and the label.
+This time I have taken it a step further: I wrote a function to get rid of the
+box and label each line.
+
+[first_notebook]: {% post_url 2020-07-27-data_science_plotting_notebook_template %}#draw-legends
+
+The function `draw_left_legend()` will draw labels on the end of each line,
+like so:
+
+[![A simple plot of the number of collisions by vehicle make in California
+with left legend][second_plot]][second_plot]
+
+[second_plot]: {{ file_dir }}/make_collision_in_time_second_version.svg
+
+I've used this legend when [_Plotting the winners of the 2019 Tour de
+France_][2019_tour] as well as the [_2020 Tour de France_][2020_tour].
+
+[2019_tour]: {% post_url 2019-08-05-2019_tour_de_france_plot %}#the-race-for-yellow
+
+## Putting It Together
+
+The [time series plotting notebook][plotting_nb] enables you to quickly plot
+your data in time with only a few lines of code. Here is the final version of
+the plot:
+
+[![An example plot from the notebook library][example]][example]
+
+[example]: {{ file_dir }}/make_collision_in_time.svg
+
+Which was produced by this short code snippet:
+
+```python
+import seaborne as sns
+
+fig, ax = setup_plot(title="Collisions by Make")
+
+pivot = plot_time_series(df, ax, date_col=DATE_COL, category_col="vehicle_make", resample_frequency="W")
```

Comment:
I assume the notebook import/namespace is handled under the hood? Sure would be nice to be explicit about `alex_lib.do_this_thing()`

Reply:
Yes, it's all in the "common" namespace.

I would love to grab the code out of the notebook, but that's rougher. I'll bug you about ideas in the future.